nhais/generate_jsonschema.py

```diff
@@ -0,0 +1,26 @@
+import json
+from jsonschema import validate
+import os
+
+import utilities.integration_adaptors_logger as log
+
+logger = log.IntegrationAdaptorsLogger(__name__)
+
+def validate_schema(request):
+    request_body = json.loads(request)
+
+    ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+    json_schema_patient = ROOT_DIR + "/json-schema-patient.json"
+
+    with open(json_schema_patient) as file:
+        data = json.load(file)
+
+    is_valid_schema = False
+
+    try:
+        validate(instance=request_body, schema=data)
```

Comment:
For current or future optimisations: if we have a unit test asserting `json-schema-patient.json` is a valid JSONSchema document then we could skip the "validate the schema" step.

> validate() will first verify that the provided schema is itself valid, since not doing so can lead to less obvious error messages and fail in less obvious or consistent ways.

> If you know you have a valid schema already, especially if you intend to validate multiple instances with the same schema, you likely would prefer using the IValidator.validate method directly on a specific validator (e.g. Draft7Validator.validate).

https://python-jsonschema.readthedocs.io/en/latest/validate/#the-basics

Reply:
I had a look into this, not sure I will be able to complete this in time for current estimation. Thoughts?